docs/model-deployment-prepackaged-server-custom.md

```diff
@@ -13,15 +13,159 @@ description: Custom Pre-packaged Server
   </div>
 </div>
 
-For better flexibility, we can also build our pre-packaged server which can load models from the model URI. Here are the recommended steps to build your first pre-packaged server.
+PrimeHub provides many [pre-packaged servers](model-deployment-prepackaged-server-intro), it might not fit in your use cases:
 
+1. You are adopting a new machine learning library, we haven't provided.
+1. You want to support a serialization format for your model, we havn't supported. 
+1. You want to customize moniotiring metrics data.
 
-1. Read the document [Package from Language Wrapper for python](model-deployment-language-wrapper-intro). This is the fundamental of a pre-packaged server.
-1. Reference existing [pre-packaged servers](https://github.com/InfuseAI/primehub-seldon-servers). Try to build and push the same pre-packaged server to your container registry (e.g. Dockerhub).
-1. In the `__init__()` method of the model class, add the parameter `model_uri` to load the model. The value is a string that contains the path where the model files are downloaded. Currently, it is always `/mnt/models` rather than the actual model URI the user configure.
+This document will show you how to customize a Python model server and focus on the `model_uri` mechanism of the PrimeHub Deployment. You could refer to existing implementation on the git repository [primehub-seldon-servers](https://github.com/InfuseAI/primehub-seldon-servers). 
 
-    ```
-    def __init__(self, model_uri):
-        super().__init__()
-        self.model = load_model(model_uri)
-    ```
+We will use the code in the [skeleton](https://github.com/InfuseAI/primehub-seldon-servers/tree/master/skeleton) server to explain concepts.
+
+## How pre-packaged server works ?
+
+If you look the [Dockerfile](https://github.com/InfuseAI/primehub-seldon-servers/blob/master/skeleton/Dockerfile) in the skeleton server:
+
+```
+FROM python:3.7-slim
+COPY ./server /app
+WORKDIR /app
+RUN pip install -r requirements.txt
+EXPOSE 5000
+EXPOSE 9000
+
+# Define environment variable
+ENV MODEL_NAME Model
+ENV SERVICE_TYPE MODEL
+ENV PERSISTENCE 0
+
+CMD exec seldon-core-microservice $MODEL_NAME --service-type $SERVICE_TYPE --persistence $PERSISTENCE --access-log
+```
+
+You will find the entrypoint is `seldon-core-microservice` with a `$MODEL_NAME` (Model). **seldon-core-microservice** plays a HTTP server to get requests from clients and delegates all model requests to your `Model`. **seldon-core-microservice** will validate input data and convert it to proper data type to the `Model`, `Model` does `predict` and sends back results.
```

Comment:
a HTTP -> an HTTP
https://www.quora.com/A-http-or-an-http-which-is-the-correct-grammatical-usage